File: RadarSDK/RadarRouteMode.m
// RadarRouteMode.m
#import "RadarRouteMode.h"

@implementation RadarRouteModeUtils

+ (NSString *)stringForMode:(RadarRouteMode)mode {
    if (mode == 0) {
        return @"unknown";
    }
    NSMutableArray *modes = [NSMutableArray array];
    if (mode & RadarRouteModeFoot) {
        [modes addObject:@"foot"];
    }
    if (mode & RadarRouteModeBike) {
        [modes addObject:@"bike"];
    }
    if (mode & RadarRouteModeCar) {
        [modes addObject:@"car"];
    }
    if (mode & RadarRouteModeTruck) {
        [modes addObject:@"truck"];
    }
    if (mode & RadarRouteModeMotorbike) {
        [modes addObject:@"motorbike"];
    }
    if (modes.count == 0) {
        return @"unknown";
    }
    return [modes componentsJoinedByString:@","];
}

@end

File: RadarSDK/RadarRouteMode.swift
//
//  RadarRouteMode.swift
//  RadarSDK
//
//  Copyright © 2019 Radar Labs, Inc. All rights reserved.
//

import Foundation

@objc @implementation extension RadarRouteModeUtils {
    override init() {
        super.init()
    }
    class func stringForMode(_ mode: RadarRouteMode) -> String {
        switch mode {
        case .foot:
            return "foot"
        case .bike:
            return "bike"
        case .car:
            return "car"
        case .truck:
            return "truck"
        case .motorbike:
            return "motorbike"
        default:
            return "unknown"
        }
    }
}
